lib/computer_player.rb

```diff
@@ -0,0 +1,37 @@
+require_relative "player"
+
+class ComputerPlayer < Player
+
+  def initialize(board, validator)
+    super(board)
+    @validator = validator
+  end
+
+  def place_ship(ship_length)
+    coordinates = get_valid_coordinates(ship_length)
+    @board.mark_board_with_ship(coordinates)
+  end
+
+  def get_valid_coordinates(ship_size)
+    coordinates = get_right_direction_coordinates(ship_size)
+    if @validator.coordinates_do_not_include_ship?(coordinates)
+      coordinates
+    else
+      get_valid_coordinates(ship_size)
+    end
+  end
+
+  def get_right_direction_coordinates(ship_size)
```

Comment:
you have a method called this in `Display` as well.  Is there any duplication that could be dried up?